src/types/Actions.ts

```diff
@@ -0,0 +1,22 @@
+import { Address } from 'viem';
+import { Prettify } from './utils';
+
+/**
+ * Actions require contract address, but as part of decorators, the address might have been passed already to the decorator.
+ *
+ * If the address was passed to the decorator, it's now optional (we still allow overrides of the address per action).
+ * If the action doesn't have any other parameters beside the contract address, then parameters can either be { contract: address } or void
+ */
+export type ActionParameters<Args, ContractName extends string, Curried extends boolean> = Prettify<
+  Curried extends false
+    ? Args & { [key in ContractName]: Address }
+    : Args extends Record<string, never>
+    ?
+        | {
+            [key in ContractName]: Address;
+          }
+        | void
```

Comment:
continuing our previous discussion, not very sure about using `void` for optional objects but approving this as it works

Reply:
It's slightly different here. If we use `void, we allow `getMaxTimeVariation()` if the sequencerInbox was passed to the decorator previously. If we use `{ [key in ContractName]?: Address }` instead, then we always require a parameter, also `getMaxTimeVariation({ sequencerInbox: undefined })` would now be a valid option.